src/app/components/ContactEmailSettingsModal.js

```diff
@@ -1,66 +1,227 @@
+    };
+
+    /**
+     * Get the keys for an email address from the API.
+     * @param {String} Email
+     * @returns {Promise<{RecipientType, MIMEType, Keys}>}
+     */
+    const getKeysFromApi = async (Email) => {
+        try {
+            // eslint-disable-next-line no-unused-vars
+            const { Code, ...data } = await api(getPublicKeys({ Email }));
+            return data;
+        } catch (error) {
+            const { data = {} } = error;
+
+            if (EMAIL_ERRORS.includes(data.Code)) {
+                return {
+                    RecipientType: TYPE_NO_RECEIVE,
+                    MIMEType: null,
+                    Keys: []
+                };
+            }
+
+            throw error;
+        }
+    };
+
+    const prepare = async () => {
+        const { contactKeys, mimeType, encrypt, scheme, sign } = properties
+            .filter(({ field, group }) => VCARD_KEY_FIELDS.includes(field) && group === emailProperty.group)
+            .reduce(
+                (acc, { field, value }) => {
+                    if (field === 'key' && value) {
+                        const [, base64 = ''] = value.split(',');
+                        const data = binaryStringToArray(decodeBase64(base64));
+
+                        if (data.length) {
+                            acc.contactKeys.push(data);
+                        }
+
+                        return acc;
+                    }
+
+                    if (field === 'x-pm-encrypt' && value) {
+                        acc.encrypt = value === 'true';
+                        return acc;
+                    }
+
+                    if (field === 'x-pm-sign' && value) {
+                        acc.sign = value === 'true';
+                        return acc;
+                    }
+
+                    if (field === 'x-pm-scheme' && value) {
+                        acc.scheme = value;
+                        return acc;
+                    }
+
+                    if (field === 'x-pm-mimetype' && value) {
+                        acc.mimeType = value;
+                        return acc;
+                    }
+
+                    return acc;
+                },
+                { contactKeys: [], mimeType: '', encrypt: false, scheme: '', sign: Sign === 1 } // Default values
+            );
+        const config = await getKeysFromApi(contactEmail.Email);
+        const internalUser = isInternalUser(config);
+        const externalUser = !internalUser;
+        const apiKeys = config.Keys.map(({ PublicKey }) => PublicKey);
+        const [unarmoredKeys, keysExpired] = await Promise.all([
+            getRawInternalKeys(config),
+            allKeysExpired(contactKeys)
+        ]);
+        const noPrimary =
+            unarmoredKeys.length &&
+            !unarmoredKeys.some((k) =>
+                contactKeys.map((value) => encodeBase64(arrayToBinaryString(value))).includes(k)
+            );
+
+        const trusted = internalUser
```

Comment:
For me `trustedFingerprints` would be a more informative name for this variable (because `trusted` sounds more like a boolean), but maybe it's called `trusted` somewhere else, and anyway this is personal taste.